contrib/golang/upstreams/http/tcp/source/upstream_request.cc

```diff
@@ -0,0 +1,529 @@
+#include "upstream_request.h"
+
+#include <cstdint>
+#include <memory>
+#include <utility>
+
+#include "envoy/upstream/upstream.h"
+
+#include "processor_state.h"
+#include "source/common/common/assert.h"
+#include "source/common/common/utility.h"
+#include "source/common/http/codes.h"
+#include "source/common/http/header_map_impl.h"
+#include "source/common/http/headers.h"
+#include "source/common/http/message_impl.h"
+#include "source/common/network/transport_socket_options_impl.h"
+#include "source/common/router/router.h"
+#include "source/extensions/common/proxy_protocol/proxy_protocol_header.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Upstreams {
+namespace Http {
+namespace Tcp {
+namespace Golang {
+
+BridgeConfig::BridgeConfig(
+    const envoy::extensions::upstreams::http::tcp::golang::v3alpha::Config proto_config, Dso::TcpUpstreamDsoPtr dso_lib)
+    : plugin_name_(proto_config.plugin_name()), so_id_(proto_config.library_id()),
+      so_path_(proto_config.library_path()), plugin_config_(proto_config.plugin_config()), dso_lib_(dso_lib){};
+
+BridgeConfig::~BridgeConfig() {
+  if (config_id_ > 0) {
+    dso_lib_->envoyGoTcpUpstreamDestroyPluginConfig(config_id_);
+  }
+}
+
+void BridgeConfig::newGoPluginConfig() {
+  ENVOY_LOG(debug, "golng http1-tcp bridge initializing golang filter config");
+  std::string buf;
+  auto res = plugin_config_.SerializeToString(&buf);
+  ASSERT(res, "SerializeToString should always successful");
+  auto buf_ptr = reinterpret_cast<unsigned long long>(buf.data());
+  auto name_ptr = reinterpret_cast<unsigned long long>(plugin_name_.data());
+
+  // plugin_name_ptr in httpConfig
+  plugin_name_ptr = name_ptr;
+  // plugin_name_len in httpConfig
+  plugin_name_len = plugin_name_.length();
+  // config_ptr in httpConfig
+  config_ptr = buf_ptr;
+  // config_len in httpConfig
+  config_len = buf.length();
+
+  config_id_ = dso_lib_->envoyGoOnTcpUpstreamConfig(this);
+
+  if (config_id_ == 0) {
+    PANIC(fmt::format("golng http1-tcp bridge failed to parse plugin config: {} {}", so_id_, so_path_));
```

Comment:
http-tcp? this also works for http2 as downstream?

Reply:
yes, so we change to call it golang http-tcp bridge.(btw, need to support trailers in the future. )